src/scene/morph-instance.js

```diff
@@ -170,25 +174,47 @@ class MorphInstance {
         return clone;
     }
 
+    _getWeightIndex(key) {
+        const index = this._weightMap.get(key);
+        if (index) {
+            return this._weights[index];
+        }
+        Debug.error('Attempting to retrieve a morph target weight using a key that doesn\'t exist.');
+    }
+
     /**
      * Gets current weight of the specified morph target.
      *
-     * @param {number} index - An index of morph target.
+     * @param {string|number} key - An identifier for the morph target. Either the weight index or the weight name
      * @returns {number} Weight.
      */
-    getWeight(index) {
+    getWeight(key) {
+        let index;
+        if (typeof key === 'string') {
+            this._getWeightIndex(key);
```

Comment:
should the result be assigned to index?

Reply:
just do `const index = (typeof key === 'string') ? this._getWeightIndex(key) : key;`

Reply:
👍